### PR TITLE
Moves all CI to GitHub Actions

### PR DESCRIPTION
**Pull request by vlj91**
- Adds the remaining steps to GitHub Actions: `lint`, `build`, and `publish`. There is a slight change in that we now always run the `lint` and `build` steps
- We also now perform the `lint`, `build`, and `publish` steps using the Node version in `.nvmrc`, rather than explicitly using Node 14 as we did in CircleCI
- Removes the CircleCI configuration

Once approved, I'll disable CircleCI prior to merging.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,3 +47,77 @@ jobs:
...
+          node-version-file: .nvmrc
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - run: yarn build
+
+      - name: Authenticate with npm registry
+        run: echo "//registry.npmjs.org/:_authToken=$NPM_PUBLISH_TOKEN" > ~/.npmrc
```
Comment (timdawborn) [inline .github/workflows/ci.yml:120]: **Issue**:

1. This should be of the form `secrets.blah`, shouldn't it?
2. This secret currently doesn't exist on the repo. There's one with a similar name but it's not this name.

Comment (vlj91) [reply .github/workflows/ci.yml:120]:
```suggestion
        run: echo "//registry.npmjs.org/:_authToken=${{ secrets. NPMJS_PUBLISH_TOKEN }}" > ~/.npmrc
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -47,3 +47,77 @@ jobs:
...
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - run: yarn build
+
+      - name: Authenticate with npm registry
+        run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPMJS_PUBLISH_TOKEN }}" > ~/.npmrc
+
+      - name: Publish to npm registry
+        run: npm publish --access=public
```
Comment (timdawborn) [inline .github/workflows/ci.yml:123]: **Suggestion**: It looks like there's a more canonical way to do this in GHA: https://docs.github.com/en/actions/publishing-packages/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

**Review by timdawborn — DISMISSED** [review]
LGTM 👍

**Review by timdawborn — APPROVED** [review]
LGTM 👍

**Pull request merged**